.github/workflows/ci-pr.yml

```diff
@@ -37,6 +37,39 @@ jobs:
         raco test -c tests/stxparse
         raco test -c tests/syntax
 
+  buildtest-linux-arm64:
+    container:
+      image: racket/racket-ci:latest
+      options: --init
+
+    runs-on: [self-hosted, linux, ARM64]
+
+    steps:
+    - uses: actions/checkout@v2
+    - name: Build
+      run: make CPUS=$(nproc) PKGS="racket-test db-test unstable-flonum-lib net-test"
```

Comment:
Should this also run Racket CS?

Reply:
I didn't want PRs to take to long to return results which is why I thought just building 3m for now (the default) was enough. I would say that for PRs, the default is enough and we run CS when CS becomes the default, by building from a snapshot of cgc or something. If there's a problem in CS, it will show when pushed and then we can revert if the problem occurs.

I think for `racket` at the moment, it doesn't really matter too much if the PR CI is super quick, but generally it's a good idea to have a quick turnaround for PRs and in-depth testing for Pushes.

Reply:
That makes sense to me for PRs, but I think we do want regular testing for pushes. Maybe the performance issues will get better when "Apple Silicon" is available, if GHA provides some.

Reply:
It may take so long because it uses BC to simulate Chez Scheme to build Chez Scheme, and BC doesn't have a JIT on AArch64. For example, on a Pi 4, it takes 15 minutes just to compile "cpnanopass.ss" using the simulated Chez Scheme. (The process is much faster for me on AArch32, since there's JIT support.)

I have been thinking about ways to improve the build time — including ways that avoid adding an AArch64 JIT to BC.

Reply:
One possibility is to use GitHub Actions caching support to cache boot files, which could then be used to avoid the long bootstrap process in most cases.

Reply:
> That makes sense to me for PRs, but I think we do want regular testing for pushes. Maybe the performance issues will get better when "Apple Silicon" is available, if GHA provides some.

Certainly, for pushes it makes sense to run even if it's slow.
I will get a Mac Mini once they are out for testing and CI, so hopefully we can use that once it's out.

Reply:
> It may take so long because it uses BC to simulate Chez Scheme to build Chez Scheme, and BC doesn't have a JIT on AArch64. For example, on a Pi 4, it takes 15 minutes just to compile "cpnanopass.ss" using the simulated Chez Scheme. (The process is much faster for me on AArch32, since there's JIT support.)
> 
> I have been thinking about ways to improve the build time — including ways that avoid adding an AArch64 JIT to BC.

Well - at the moment those 1h30m is not even building CS. That's just CGC and 3m. If we tried CS would definitely be a bit more. And we'll see how much more once we get it running in push. I will get this added tomorrow.

Reply:
> One possibility is to use GitHub Actions caching support to cache boot files, which could then be used to avoid the long bootstrap process in most cases.

That might be useful for the push CI - can we really use old boot files, to build a recent CS?

Reply:
Note that in this case building CS directly from CGC might make things faster, instead of the intermediate 3m step

Reply:
Ah I misunderstood the 1h30m. It looks like a CGC-them-3m build takes 50 minutes, and then testing is 40 minutes. During the build, roughly 20 minutes is `raco setup`. I bet a lot of the build time is compiling the cified "cstartup.inc".